### PR TITLE
info memory command extended to show peak memory on memkind

### PR DESCRIPTION
**Pull request by karpiop**
This change is [Reviewable](https://reviewable.io/reviews/pmem/redis/66)

### CODE REVIEW

File: src/server.h [hunk 1]
```diff
@@ -952,6 +952,7 @@ struct redisServer {
     long long stat_active_defrag_key_hits;  /* number of keys with moved allocations */
     long long stat_active_defrag_key_misses;/* number of keys scanned and not moved */
     size_t stat_peak_memory;        /* Max used memory record */
+    size_t stat_peak_memory_memkind;/* Max used memkind memory record */
```
Comment (michalbiesek) [inline src/server.h:955]: You don't want to initialize this value in initServer method ?

Comment (karpiop) [reply src/server.h:955]: That's good point. Fixed.

File: src/server.c [hunk 2]
```diff
@@ -2981,7 +2986,10 @@ sds genRedisInfoString(char *section) {
             "used_memory_rss_human:%s\r\n"
             "used_memory_peak:%zu\r\n"
             "used_memory_peak_human:%s\r\n"
+            "used_memory_peak_memkind:%zu\r\n"
```
Comment (michalbiesek) [inline src/server.c]: check - order of new specifiers ( lines 2989 2991 and 2992 ) with order of new variables ( lines 3017-3019 ). Line 2990 ?

File: src/server.c [hunk 3]
```diff
@@ -3006,6 +3014,9 @@ sds genRedisInfoString(char *section) {
             server.stat_peak_memory,
             peak_hmem,
             mh->peak_perc,
+            server.stat_peak_memory_memkind,
+            peak_memkind_hmem,
+            (float)memkind_malloc_used*100/server.stat_peak_memory_memkind,
```
Comment (michalbiesek) [inline src/server.c]: Please consider division by zero here - if server.stat_peak_memory_memkind is 0 You call here Undefined Behaviour

Comment (KFilipek) [reply src/server.c]: I never recommend division by zero :P

Comment (KFilipek) [reply src/server.c]: server.stat_peak_memory_memkind == 0 ? 0.0 : (float)memkind_malloc_used*100/server.stat_peak_memory_memkind,

**Comment by michalbiesek** [comment]
:lgtm:

---

Reviewed 1 of 2 files at r1, 1 of 1 files at r3.
Review status: :shipit: complete! all files reviewed, all discussions resolved

---



*Comments from [Reviewable](https://reviewable.io/reviews/pmem/redis/66#-:-LEnGi7a7zIolUMcB7n8:bnfp4nl)*

**Comment by KFilipek** [comment]
:lgtm:

---

Reviewed 1 of 2 files at r1, 1 of 1 files at r3.
Review status: :shipit: complete! all files reviewed, all discussions resolved

---



*Comments from [Reviewable](https://reviewable.io/reviews/pmem/redis/66#-:-LEo3VbE4HDA-zkzPwWG:bnfp4nl)*

**Pull request merged**